plugins/inputs/prometheus_scraper/metrics_type_handler.go

```diff
@@ -40,19 +40,19 @@ func normalizeMetricName(name string, suffixes []string) string {
 }
 
 func (pm *PrometheusMetric) isCounter() bool {
-	return pm.metricType == textparse.MetricTypeCounter
+	return pm.metricType == string(textparse.MetricTypeCounter)
```

Comment:
Here is a breaking change from dependency, instead of returning string, it's now returning a enum of **MetricType**, note this is **not** an interface, so I'm casting it to string to keep it work with the PrometheusMetric struct.